### PR TITLE
ADD: Keychain clear for iOS [WIP]

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/uq5f1yb8pc0b8bxm52q20n8xkg

[download apk](http://filestorage.bluewallet.io:1488/1590026941.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/jqqz198mceumw91n5ku0ka5bp8

[download apk](http://filestorage.bluewallet.io:1488/1590038175.apk)

File: BlueApp.js [hunk 1]
```diff
@@ -11,12 +13,13 @@ let BlueElectrum = require('./BlueElectrum'); // eslint-disable-line
...
 /** @type {AppStorage} */
 const BlueApp = new AppStorage();
+let attempt = 0;
 
 async function startAndDecrypt(retry) {
   console.log('startAndDecrypt');
   if (BlueApp.getWallets().length > 0) {
     console.log('App already has some wallets, so we are in already started state, exiting startAndDecrypt');
-    return;
+    return false;
```
Comment (Overtorment) [inline BlueApp.js:22]: feels like unnecessary change

File: BlueApp.js [hunk 2]
```diff
@@ -58,11 +60,18 @@ async function startAndDecrypt(retry) {
     if (hadToRefresh && noErr) {
       await BlueApp.saveToDisk(); // caching
     } */
+    return true;
```
Comment (Overtorment) [inline BlueApp.js:63]: feels like unnecessary change

File: BlueApp.js [hunk 3]
```diff
@@ -58,11 +60,18 @@ async function startAndDecrypt(retry) {
...
 
   if (!success && password) {
     // we had password and yet could not load/decrypt
-    return startAndDecrypt(true);
+    attempt++;
+    if (attempt < 10 || Platform.OS !== 'ios') {
+      return startAndDecrypt(true);
+    } else {
+      attempt = 0;
+      Biometric.showKeychainWipeAlert();
```
Comment (Overtorment) [inline BlueApp.js:73]: you should await for result here and then do `return startAndDecrypt(true);`  or `return startAndDecrypt(false);` based on whether user decided to clear his keychain (so no password is needed anymore) or decided to try again with the password. obviously that should be ios path. android path is way simpler.
that way you can get rid of new EV event and a bunch of other unnecessary complexity.

File: UnlockWith.js [hunk 4]
```diff
@@ -5,13 +5,14 @@ import Biometric from './class/biometrics';
...
 /** @type {AppStorage} */
-
+const EV = require('./events');
 const BlueApp = require('./BlueApp');
 
 export default class UnlockWith extends Component {
   state = { biometricType: false, isStorageEncrypted: false, isAuthenticating: false };
 
   async componentDidMount() {
+    EV(EV.enum.KEYCHAIN_CLEARED, this.unlockWithKey);
```
Comment (Overtorment) [inline UnlockWith.js:15]: feels like unnecessary change

**Comment by Overtorment** [comment]
@marcosrdz are you going to work on this?

**Comment by marcosrdz** [comment]
> @marcosrdz are you going to work on this?

Yup. Gotta fix some stuff.
